### ISSUE TITLE
Investigate issue with router restarts making data fail to send on reconnection

### ISSUE THREAD

**Issue by ajayyy** [uorocketry/rocket-code-2020#138]
At the cold flow test, we had an issue where reconnecting after the connection failed would sometimes cause the connection to succeed, while recieving no data. The ground station could still send control bits, but it was not recieving any data.

The cause is unknown, but it could have something to do with:
- too many connections (tested with ~6 with no issue though)
- zombie connections sticking around after router restarts
- router restarting
- router getting too low of voltage from inverter
- pi getting too low voltage from inverter
or something else...

Things ruled out:
- static IP
- for loop

***

Looking at the raspberry pi logs could be helpful too.

**Comment by ajayyy** [comment]
Reopening until we can reproduce it

**Comment by ajayyy** [comment]
We managed to reproduce this and the fix does work. To reproduce this, we connected two ground stations to the rocket code, then unplugged one. After a few minutes, the remaining ground station stops receiving data and the code gets frozen when trying to write to the TCP socket.

This is probably due to a buffer filling up, causing the os to require the program to wait until the rest of the data has been sent.
